### PR TITLE
Switch to a new progress event for build status

### PR DESCRIPTION
**Pull request by tmeasday**
Using the new `onTaskProgress` option, switch to a more full featured `BUILD_PROGRESS` event https://github.com/chromaui/chromatic-cli/pull/805.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.42--canary.55.36a4f81.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@0.0.42--canary.55.36a4f81.0
  # or 
  yarn add @chromaui/addon-visual-tests@0.0.42--canary.55.36a4f81.0
  ```
</details>

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -70,19 +69,42 @@ async function serverChannel(
...
-            started = true;
+          if (step === "upload" && ctx.isolatorUrl) {
+            newStep = "verify";
           }
+          if (["build", "upload"].includes(step) && ctx.build) {
+            newStep = "snapshot";
+          }
+          if (ctx.build?.status !== "IN_PROGRESS") {
+            newStep = "complete";
+          }
```
Comment (tmeasday) [inline src/index.ts:84]: Not quite sure about this logic, @ghengeveld do you have a better idea?

Comment (ghengeveld) [reply src/index.ts:84]: I'm not digging the `let newStep`. I'd rather have a function that returns the new step.

As you know I don't like `let` at all. I can sort of _let_ it slide when it's used for memory storage over time (such as `step` above), but not when it's used with a series of `if` statements to (synchronously) determine its final value.

**Review by ghengeveld — CHANGES_REQUESTED** [review]

File: src/screens/VisualTests/BuildStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export function BuildStatus({}) {
+  return <div>Build Status</div>;
+}
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply src/screens/VisualTests/BuildStatus.tsx:5]: Yes, there is a second PR that actually implements the UI: https://github.com/chromaui/addon-visual-tests/pull/56

**Comment by tmeasday** [comment]
I'm going to take a second pass over this once I've figured out the server->manager/preview use addon state (as I will store this build progress information in addon state). 

@weeksling I will leave this PR so you can base any UI work on it but be aware I am going to change quite a lot of things later (probably in a distinct PR).

**Comment by tmeasday** [comment]
This was merged as part of https://github.com/chromaui/addon-visual-tests/pull/69 with a lot of changes.

**Pull request closed without merging**